pkg/parquetquery/iters.go

```diff
@@ -710,6 +715,34 @@ func (c *SyncIterator) seekPages(seekTo RowNumber, definitionLevel int) (done bo
 	return false, nil
 }
 
+func (c *SyncIterator) seekRows(to RowNumber, definitionLevel int) {
+	// only skip if to is ahead of curr
+	if CompareRowNumbers(definitionLevel, to, c.curr) > 0 {
+		// skips are calculated off the start of the page
+		rowSkip := to[0] - c.currPageMin[0]
+		if rowSkip > 5 { // made up threshold worth skipping
+			// reslice the page to jump directly to the desired row number
+			pg := c.currPage.Slice(rowSkip-1, c.currPage.NumRows())
+			pq.Release(c.currPage) //- can i release here? i think the internal buffers are  are preserved, so no
```

Comment:
setPage is slightly askew of what i need here (which is why i didn't use it). i might be able to pass a new param? or game it some by calling setPage and then chaing some state vars afterward? but this felt cleaner.

the big difference is that setPage manipulates c.curr to "skip" the current page which this logic does not want to do:
https://github.com/grafana/tempo/blob/main/pkg/parquetquery/iters.go#L803

i also noticed this which i don't quite get:
https://github.com/grafana/tempo/blob/main/pkg/parquetquery/iters.go#L815

it seems like we're clearing the buffer only if the incoming page is nil? shouldn't we clear the buffer if the incoming page is not nil? so that it doesn't reuse the buffer for the new page?